src/main/java/seedu/address/model/person/Nric.java

```diff
@@ -27,7 +27,7 @@ public class Nric {
     public Nric(String nric) {
         requireNonNull(nric);
         checkArgument(isValidNric(nric), MESSAGE_CONSTRAINTS);
-        value = nric;
+        value = nric.toUpperCase();
```

Comment:
Storing in the client list will still be in upper-case regardless of the case given by the user.